client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -160,19 +162,41 @@ private Checkpoint completeCheckpoint(String checkpointName, StateSynchronizer<R
     }
 
     @Override
-    public void resetReadersToCheckpoint(Checkpoint checkpoint) {
+    public void resetReaders(final Checkpoint checkpoint) {
+        resetReaders(checkpoint.asImpl().getPositions());
+    }
+
+    @Override
+    public void resetReaders(final Map<Stream, StreamCut> streamCuts) {
         @Cleanup
         StateSynchronizer<ReaderGroupState> synchronizer = createSynchronizer();
+        synchronizer.fetchUpdates();
+
+        //ensure that streamCut for all the streams managed by the ReaderGroup are present.
+        Preconditions.checkArgument(validateStreamCuts(streamCuts.values(), synchronizer.getState().getStreamNames()),
+                "StreamCuts for streams managed by the ReaderGroup need to be provided");
+
+        //reset the Readers to the StreamCut.
         synchronizer.updateState(state -> {
             ReaderGroupConfig config = state.getConfig();
             Map<Segment, Long> positions = new HashMap<>();
-            for (StreamCut cut : checkpoint.asImpl().getPositions().values()) {
+            for (StreamCut cut : streamCuts.values()) {
                 positions.putAll(cut.getPositions());
             }
             return Collections.singletonList(new ReaderGroupStateInit(config, positions));
         });
     }
 
+    /*
```

Comment:
I understand that you are avoiding to do a javadoc here because it is a private method, but it looks better if you format the block like this:

```
     /*
      *   Validate the following
      *       - provided stream cuts are unique and has no duplicates.
      *       - stream cuts for all the streams managed by the reader group are present.
      */
```

Reply:
updated the code.

Reply:
Ok.